examples/blend.cpp

```diff
@@ -119,33 +206,70 @@ class Blend
             memset(&m_dest->crop, 0, sizeof(VideoRect));
             m_scaler->process(frame, m_dest);
 
-            PERF_START(blend);
-            //blend it
-            for (uint32_t i = 0; i < m_blendSurfaces.size(); i++) {
-                m_blendBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
-                //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+            if (m_bBlend) {
+                PERF_START(blend);
+                //blend it
+                for (uint32_t i = 0; i < m_blendSurfaces.size(); i++) {
+                    m_blendBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+                    //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+
+                    SharedPtr<VideoFrame>& src = m_blendSurfaces[i];
+                    m_blender->process(src, m_dest);
+                }
+                PERF_STOP(blend);
+            }
+
+            if (m_bOsd) {
+                PERF_START(osd);
+                //osd
+                for (uint32_t i = 0; i < m_osdSurfaces.size(); i++) {
+                    m_osdBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+                    //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
 
-                SharedPtr<VideoFrame>& src = m_blendSurfaces[i];
-                m_blender->process(src, m_dest);
+                    SharedPtr<VideoFrame>& src = m_osdSurfaces[i];
+                    m_osd->process(src, m_dest);
+                }
+                PERF_STOP(osd);
             }
-            PERF_STOP(blend);
 
-            PERF_START(osd);
-            //osd
-            for (uint32_t i = 0; i < m_osdSurfaces.size(); i++) {
-                m_osdBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
-                //printf("(%d, %d, %d, %d)\r\n", m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
+            if (m_bMosaic) {
+                PERF_START(mosaic);
+                //mosaic
+                for (uint32_t i = 0; i < m_mosaicBumpBoxes.size(); i++) {
+                    m_mosaicBumpBoxes[i]->getPos(m_dest->crop.x, m_dest->crop.y, m_dest->crop.width, m_dest->crop.height);
 
-                SharedPtr<VideoFrame>& src = m_osdSurfaces[i];
-                m_osd->process(src, m_dest);
+                    SharedPtr<VideoFrame> empty;
+                    m_mosaic->process(empty, m_dest);
```

Comment:
should call it like this m_mosaic->process(m_dest, m_dest);.So you did not create empty surface. for mosic operation m_dest is src and dest.